algorithms/src/fft/polynomial/sparse.rs

```diff
@@ -18,12 +18,11 @@
 
 use crate::fft::{EvaluationDomain, Evaluations, Polynomial};
 use snarkvm_fields::{Field, PrimeField};
-use snarkvm_utilities::serialize::*;
 
 use std::{collections::BTreeMap, fmt};
 
 /// Stores a sparse polynomial in coefficient form.
-#[derive(Clone, PartialEq, Eq, Hash, Default, CanonicalSerialize, CanonicalDeserialize)]
+#[derive(Clone, PartialEq, Eq, Hash, Default)]
```

Comment:
side note: it might be a good idea to check if any other objects can be "trimmed" in a similar fashion for faster compilation

Reply:
Added a thumbsup! https://github.com/rust-lang/rust/issues/50927